MovieApp/MovieApp/Views/SearchMovies/ViewController/SearchMoviesViewController.swift

```diff
@@ -1,117 +1,82 @@
+    // MARK: - Properties
+    
+    @IBOutlet weak var tableView: UITableView!
+    private var dataSource: MoiveListDatsSource?
+    private var delegate: MovieListDelegate?
+    
+    private var viewModel: SearchMovieViewModelProtocol?
+    private let searchController = UISearchController(searchResultsController: nil)
+    var spinner: SpinnerViewController = SpinnerViewController()
+    
+    struct Actions {
+        
+        var openMovie: (_ movieID: String) -> Void
     }
-    let movie = viewModel?.movie(at: indexPath.row)
-    print("🤡 \(movie)")
-    cell.setUp(from: movie)
-    return cell
-  }
-  
-  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-    let movie = viewModel?.movie(at: indexPath.row)
-    self.actions?.openMovie(movie?.imdbID ?? "")
-  }
-}
-
-extension SearchMoviesViewController: SearchDelegate {
-
-  func reloadTableView() {
-    DispatchQueue.main.async { [weak self] in
-      self?.tableView?.reloadData()
+    
+    var actions: Actions?
+    
+    convenience init(viewModel: SearchMovieViewModelProtocol) {
+        self.init(nibName: nil, bundle: nil)
+        
+        self.viewModel = viewModel
+        dataSource = MoiveListDatsSource(viewModel: viewModel)
+        delegate = MovieListDelegate(viewModel: viewModel)
+    }
+    
+    // MARK: - Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        configureSearchBar()
+        configureNavBar()
+        setupTableView()
+        fetchMovies(by: "Pulp Fiction")
+    }
+    
+    // MARK: - Private
+    
+    private func setupTableView() {
+        actions.do { actions in
+            delegate?.actions = .init(openMovie: actions.openMovie)
+        }
+        MovieListStyle.baseMovieListStyle(tableView)
+        tableView.dataSource = dataSource
+        tableView.delegate = delegate
+    }
+    
+    private func configureNavBar() {
+        let localizedTitle = "Search Movies".localized()
+        navigationItem.title = localizedTitle
+    }
+    
+    private func fetchMovies(by title: String) {
+        viewModel?.onDataLoaded = { [weak self] in
+            DispatchQueue.main.async {
+                self?.hideSpinner()
+                self?.tableView.reloadData()
+            }
+        }
+        showSpinner()
+        viewModel?.fetchMovies(by: title)
+    }
+    
+    private func configureSearchBar() {
+        navigationItem.searchController = searchController
+        navigationItem.hidesSearchBarWhenScrolling = true
+        searchController.searchBar.delegate = self
     }
-  }
 }
 
-extension SearchMoviesViewController: UISearchBarDelegate {
+// MARK: - Extensions
 
-  func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
-    fetchMovies(by: searchBar.text ?? "")
-  }
+extension SearchMoviesViewController: UISearchBarDelegate {
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        fetchMovies(by: searchBar.text ?? "")
```

Comment:
This should work like: viewModel.searchMovie(with text:), viewModel.didActivateSearch(with text:) etc.